### PR TITLE
[SYCL] Add sub-group functions emulation for vector of doubles.

### PR DESCRIPTION
**Pull request by maksimsab**
https://github.com/intel/llvm-test-suite/pull/1603

### CODE REVIEW

**Comment by maksimsab** [comment]
https://github.com/intel/llvm-test-suite/pull/1603

**Review by Pennycook — APPROVED** [review]
LGTM

**Comment by maksimsab** [comment]
@intel/llvm-gatekeepers 
Could you please merge it?

**Comment by dm-vodopyanov** [comment]
> @intel/llvm-gatekeepers Could you please merge it?

```
Code owner review required
Waiting on code owner review from intel/llvm-reviewers-runtime.
```

@maksimsab: @intel/llvm-reviewers-runtime review is required before merging.

File: sycl/include/sycl/detail/spirv.hpp [hunk 1]
```diff
@@ -498,22 +498,44 @@ AtomicMax(multi_ptr<T, AddressSpace, IsDecorated> MPtr, memory_scope Scope,
...
+struct VecTypeIsProhibitedForShuffleEmulation
+    : bool_constant<
+          (detail::get_vec_size<T>::size > 1) &&
+          TypeIsProhibitedForShuffleEmulation<vector_element_t<T>>::value> {};
+
 template <typename T>
 using EnableIfNativeShuffle =
-    detail::enable_if_t<detail::is_arithmetic<T>::value, T>;
-#else
+    detail::enable_if_t<detail::is_arithmetic<T>::value &&
```
Comment (dm-vodopyanov) [inline sycl/include/sycl/detail/spirv.hpp:521]: `std::enable_if_t`, ```std::is_arithmetic<T>```?

File: sycl/include/sycl/detail/spirv.hpp [hunk 2]
```diff
@@ -498,22 +498,44 @@ AtomicMax(multi_ptr<T, AddressSpace, IsDecorated> MPtr, memory_scope Scope,
...
+                            !VecTypeIsProhibitedForShuffleEmulation<T>::value,
+                        T>;
+
+template <typename T>
+using EnableIfVectorShuffle =
+    detail::enable_if_t<VecTypeIsProhibitedForShuffleEmulation<T>::value, T>;
```
Comment (dm-vodopyanov) [inline sycl/include/sycl/detail/spirv.hpp:527]: ```std::enable_if_t```?

File: sycl/include/sycl/detail/spirv.hpp (hunk 1 above)
Comment (maksimsab) [reply sycl/include/sycl/detail/spirv.hpp:521]: Although they have similar names, `detail::is_arithmetic` accepts `vec` types that I need.

`std::enable_if_t` probably could be used.

**Review by dm-vodopyanov — APPROVED** [review]

**Comment by dm-vodopyanov** [comment]
/verify with https://github.com/intel/llvm-test-suite/pull/1603

**Pull request merged**